File: RayTracerVBSingleProj/Camera.vb
Public Class Camera
    Public ReadOnly Pos As Vector3
    Public ReadOnly Forward As Vector3
    Public ReadOnly Up As Vector3
    Public ReadOnly Right As Vector3

    Public Sub New(ByVal pos As Vector3, ByVal lookAt As Vector3)
        Forward = (lookAt - pos).Normalized()
        Dim down = New Vector3(0, -1, 0)
        Right = Forward.CrossProduct(down).Normalized() * 1.5
        Up = Forward.CrossProduct(Right).Normalized() * 1.5
        Me.Pos = pos
    End Sub

    Private Function RecenterX(ByVal x As Double, ByVal screenWidth As Integer) As Double
        Return (x - (screenWidth / 2.0)) / (2.0 * screenWidth)
    End Function

    Private Function RecenterY(ByVal y As Double, ByVal screenHeight As Integer) As Double
        Return -(y - (screenHeight / 2.0)) / (2.0 * screenHeight)
    End Function

    Public Function GetPoint(ByVal x As Double, ByVal y As Double, ByVal screenWidth As Integer, ByVal screenHeight As Integer) As Vector3
        Return (Forward + RecenterX(x, screenWidth) * Right + RecenterY(y, screenHeight) * Up).Normalized()
    End Function
End Class

File: RayTracerCSSingleProj/Camera.cs
namespace RayTracer
{
    public class Camera
    {
        public Vector3 Pos;
        private Vector3 Forward;
        private Vector3 Up;
        private Vector3 Right;

        public Camera(Vector3 pos, Vector3 lookAt)
        {
            Forward = (lookAt - pos).Normalized();
            var down = new Vector3(0, -1, 0);
            Right = Forward.CrossProduct(down).Normalized() * 1.5;
            Up = Forward.CrossProduct(Right).Normalized() * 1.5;
            Pos = pos;
        }

        private double RecenterX(double x, int screenWidth)
        {
            return (x - (screenWidth / 2.0)) / (2.0 * screenWidth);
        }
        private double RecenterY(double y, int screenHeight)
        {
            return -(y - (screenHeight / 2.0)) / (2.0 * screenHeight);
        }

        public Vector3 GetPoint(double x, double y, int screenWidth, int screenHeight)
        {
            return (Forward + RecenterX(x, screenWidth) * Right + RecenterY(y, screenHeight) * Up).Normalized();
        }
    }
}
